### ISSUE TITLE
Phantasy Star ii md+ compatible with "Phantasy Star II Modernization" hack

### ISSUE THREAD

**Issue by pepodmc3** [Infinest/Phantasy-Star-II-MD-Plus#2]
Hello.

I want to ask you if you can make a compatible md+ patch for the "Phantasy Star II Modernization" patch for this game:

https://www.romhacking.net/hacks/1404/

It adds a lot of improvements for the game, but its not compatible with yout md+ patch.

Is this feasible or would be too much work?

thanks for your hard work in the md+ patches!

**Comment by Infinest** [comment]
Hey there,

a while ago someone else came to me with the same request but for another improvement patch.
The issue with these patches is that they touch quite a lot of the game's code and the code for the music driver as well.

Taking a quick look, the patch you've linked seems to have a lot less collisions with my patch compared to the other one .
However, that doesn't mean the rest of their code changes  don't interfere with the code i've modified.

Making the two patches compatible is definitely feasible but personally i'm not very motivated to do so.
Especially because none of these patches provide any source code which would probably make this endeavour really easy.
Without the source code you have to reverse engineer things all over again.

I might take another look to see if working out just the collisions will fix it.
If not, i'm probably not going to put any more time into it.

**Comment by pepodmc3** [comment]
> Hey there,
> 
> a while ago someone else came to me with the same request but for another improvement patch. The issue with these patches is that they touch quite a lot of the game's code and the code for the music driver as well.
> 
> Taking a quick look, the patch you've linked seems to have a lot less collisions with my patch compared to the other one . However, that doesn't mean the rest of their code changes don't interfere with the code i've modified.
> 
> Making the two patches compatible is definitely feasible but personally i'm not very motivated to do so. Especially because none of these patches provide any source code which would probably make this endeavour really easy. Without the source code you have to reverse engineer things all over again.
> 
> I might take another look to see if working out just the collisions will fix it. If not, i'm probably not going to put any more time into it.


No problem, i suspected that would be a lot of extra work for a game that already has a md+ patch, making the work twice would be pointless when you can expend that time in another project.

one more question, would you be more motivated to do a patch for a game if you see that there is a complete soundpack compiled for it? 

I compiled the soundpack of the comix zone patch you made months ago (rellik normalized the volume of the tracks and made the loops).
Because i compiled more soundpacks in the last 3 months, one for thunder force 4 that has 48 tracks. Only 4 covers missing which 2 are omakes (tracks not used in the game) so, only 2 covers tracks missing.

I have others soundpacks nearly complete too , but tell me if you consider the idea, i can send you a link with the soundpacks in one or two days to check it if you want

**Comment by Infinest** [comment]
Yes, i know that you made the Comix Zone pack. Thank you for making it.
I've had another game in the oven for a while now but i'm kind of taking a break at the moment.

However, depending on the game i might be interested in trying my hand at making a patch, yes. Thunder Force 4 for example, would be something i'm interesting in working on if you can complete the pack. I'm not gonna make any promises on actually finishing it, though. ^^

I have now taken a look at the patch you linked me and after moving my code around and fixing some offsets it seems to boot and i can get ingame. Only tested it on Neodev's mame build and not on real hardware though, so you'll have to test it further.
I'll attach the ips patch. First apply the improvement patch to your ROM and then the patch attached here.
[patch.zip](https://github.com/Infinest/Phantasy-Star-II-MD-Plus/files/7359988/patch.zip)

**Comment by pepodmc3** [comment]
Thanks a lot man for fixing the ps2 patch!! I will test it further.

About the new patches,  i will try to contact some of the youtubers that make covers to see if i can get them to make the missing covers.

And look further everywhere. When i get the soundtrack complete i will tell you.

**Comment by Infinest** [comment]
Sure thing and no problem.
I'll leave this open for now. I'd appreciate updates if you find anything/nothing wrong with the patch.
Once i can be reasonably sure that the patch is all good, i'll update the repo with my edits.

**Comment by pepodmc3** [comment]
I played until the first half of the game and no problems, need to finish it still, my computer broke so i need to find the save in my old hard drive.

About the complete arrange soundtracks, i have one for castlevania bloodlines. i compiled a very good soundpack for it .

I had a (i consider) good idea for the patch .

All stages have at least one miniboss, in which the music of the stage stops, starts the miniboss music, you defeat the miniboss, and the stage track starts again.

I found good covers in two styles, orchestra and rock. And as i couldnt decide which one choose for the stages, i was thinking in, start all the stages with the orchestra version, and when the stage tracks starts to sound again after defeating a miniboss, play the rock version of the track. With this the soundtrack will be very varied.

There are more than one miniboss in later stages, but the idea would be repeat the rock track if necessary.

ArcadeTV was going to make it, but now that he had problems and stopped making patches,  i dont think that it will happen.


I can give you a link with the music pack so you can check it. i searched everywhere for covers because for some specific tracks, it was difficult to find. All the tracks has name of the stage so you can get the double track idea i had quickly.

Only 3 covers missing from a 31 tracks soundtrack:

"stage clear" and "all clear" (The two are very short tracks after finishing stage) 
"death" theme (when you lose a life, very short). 

Using the original tracks for these 3 would be better because these three tracks are so short, that its difficult that someone will spend time making cover for those.

The track "six servants of the devil" has a loud piano in some parts, but i asked the author to make a version with a less strong piano.

https://www.youtube.com/watch?v=LZtJZhAViuI

https://drive.google.com/file/d/102_5di-WYrCh9KpjgbGtYzYL6sBKaxQg/view?usp=sharing
